### ISSUE TITLE
function or associated item not found in

### ISSUE THREAD

**Issue by hayathms** [jmacdonald/amp#109]
I am not able to install amp latest, however I am able to install 0.4.1
rustc 1.29.1

below is error when I run ->

`cargo --verbose install amp

769f/out`
error[E0599]: no function or associated item named `from_str` found for type `scribe::buffer::Distance` in the current scope
  --> src/models/application/modes/jump/mod.rs:80:32
   |
80 |                 let distance = Distance::from_str(&subtoken.lexeme);
   |                                ^^^^^^^^^^^^^^^^^^ function or associated item not found in `scribe::buffer::Distance`

error[E0599]: no function or associated item named `from_str` found for type `scribe::buffer::Distance` in the current scope
   --> src/models/application/modes/jump/mod.rs:152:40
    |
152 |                         let distance = Distance::from_str(&subtoken.lexeme);
    |                                        ^^^^^^^^^^^^^^^^^^ function or associated item not found in `scribe::buffer::Distance`

error[E0599]: no function or associated item named `from_str` found for type `scribe::buffer::Distance` in the current scope
  --> src/models/application/modes/search.rs:29:24
   |
29 |         let distance = Distance::from_str(&query);
   |                        ^^^^^^^^^^^^^^^^^^ function or associated item not found in `scribe::buffer::Distance`

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0599`.
error: failed to compile `amp v0.5.0`, intermediate artifacts can be found at `/tmp/cargo-installPx4Wpj`

Caused by:
  Could not compile `amp`.

Caused by:
  process didn't exit successfully: `rustc --crate-name amp src/lib.rs --crate-type lib --emit=dep-info,link -C opt-level=3 -C metadata=67fb7e66f6a3ef3b -C extra-filename=-67fb7e66f6a3ef3b --out-dir /tmp/cargo-installPx4Wpj/release/deps -L dependency=/tmp/cargo-installPx4Wpj/release/deps --extern app_dirs=/tmp/cargo-installPx4Wpj/release/deps/libapp_dirs-6d50622f917d078a.rlib --extern bloodhound=/tmp/cargo-installPx4Wpj/release/deps/libbloodhound-cc32f0217fce719e.rlib --extern clipboard=/tmp/cargo-installPx4Wpj/release/deps/libclipboard-821b347481808c14.rlib --extern error_chain=/tmp/cargo-installPx4Wpj/release/deps/liberror_chain-56534598579f6da6.rlib --extern fragment=/tmp/cargo-installPx4Wpj/release/deps/libfragment-56593b5db3ed4814.rlib --extern git2=/tmp/cargo-installPx4Wpj/release/deps/libgit2-69a4ce64c2d3d689.rlib --extern libc=/tmp/cargo-installPx4Wpj/release/deps/liblibc-c2cf6c5253c505f3.rlib --extern luthor=/tmp/cargo-installPx4Wpj/release/deps/libluthor-86a80c7f9d64805a.rlib --extern pad=/tmp/cargo-installPx4Wpj/release/deps/libpad-c7d1a1debebe39ca.rlib --extern regex=/tmp/cargo-installPx4Wpj/release/deps/libregex-719f1a2fedc52553.rlib --extern rustbox=/tmp/cargo-installPx4Wpj/release/deps/librustbox-11cb824f41fc2b46.rlib --extern scribe=/tmp/cargo-installPx4Wpj/release/deps/libscribe-33b84d85bd99bc57.rlib --extern smallvec=/tmp/cargo-installPx4Wpj/release/deps/libsmallvec-469b16ab340d9fb3.rlib --extern syntect=/tmp/cargo-installPx4Wpj/release/deps/libsyntect-a27f96dae07891bf.rlib --extern termion=/tmp/cargo-installPx4Wpj/release/deps/libtermion-10267157333d4b9f.rlib --extern unicode_segmentation=/tmp/cargo-installPx4Wpj/release/deps/libunicode_segmentation-c0d4432fb49c39bf.rlib --extern yaml_rust=/tmp/cargo-installPx4Wpj/release/deps/libyaml_rust-b8bc8ab6e40d9398.rlib --cap-lints allow -L native=/tmp/cargo-installPx4Wpj/release/build/backtrace-sys-6df14f2d868a98bd/out -L native=/tmp/cargo-installPx4Wpj/release/build/libgit2-sys-95690e0d097a6219/out/lib -L native=/tmp/cargo-installPx4Wpj/release/build/curl-sys-3b03448629a23ae7/out/build -L native=/usr/lib/x86_64-linux-gnu -L native=/tmp/cargo-installPx4Wpj/release/build/libssh2-sys-e60307063778cc43/out/build -L /tmp/cargo-installPx4Wpj/release/build/termbox-sys-86c35bec4fe3ddaf/out/lib -L native=/tmp/cargo-installPx4Wpj/release/build/miniz-sys-f9feca5c7202db2c/out -L native=/tmp/cargo-installPx4Wpj/release/build/onig_sys-d00ab1bf707d769f/out` (exit code: 1)
`

**Comment by jmacdonald** [comment]
Hi @hayathms! Thanks for filing an issue. This is due to a breaking change in `scribe`, and unfortunately, crates.io doesn't support deterministic builds (it doesn't use the `Cargo.lock` file); sorry about that.

TL;DR: I've issued a new `0.5.1` release of Amp that's compatible with the latest version of scribe, which should fix this issue. Can you please try installing it again?

**Comment by hayathms** [comment]
@jmacdonald It installed like butter, its great editor, please keep up the great work, everyone loved it in office when i pressed "f" and showed them Time Machine.,

**Comment by jmacdonald** [comment]
@hayathms glad to hear it; thanks for the kind words! ❤️
